WalletWasabi/Blockchain/TransactionBuilding/SmartCoinSelector.cs

```diff
@@ -65,21 +65,20 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
 
 		var coinsInBestClusterByScript = bestCoinCluster
 			.GroupBy(c => c.ScriptPubKey)
-			.Select(group => new { Coins = group });
+			.Select(group => (ScriptPubKey: group.Key, Coins: group.ToList()))
+			.ToImmutableList();
 
-		var coinsToSpend = new List<SmartCoin>();
+		var coinsGroup = coinsInBestClusterByScript.Count < 10
```

Comment:
Are you sure 10 is ok on all computers?

Reply:
Sorry, I should have explained the change. Once we have the a set of coins, grouped by scripPubKey, in the the `coinsInBestClusterByScript` the question is, which of those coins should be selected? 

Before this PR the logic was to take coins until they were enough. After this PR the logic is find the best combination. The complex part in how to compute the combinations because we have the best possible solution that is find all the combinations that only works when there are only a few coins but then, what to do in case there are lots of coins in that cluster well I will prepare an example. 

> Are you sure 10 is ok on all computers?

No, I am not sure but that's the number we are using a few lines o code above so, it seems we assumed it was okay. Anyway, it would be good to play we this a bit more.

I am switching it back to draft until I have better answers and at least one example.